File: frontends/main/src/app/configurator/services/api/timeline-graph-grants.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({providedIn: 'root'})
export class TimelineGraphGrantsService {
	constructor(private http: HttpClient) {}

	getTimelineGraphGrants(projectId: string): Observable<Record<string, string[]>> {
		return this.http.get<Record<string, string[]>>(`/api/superuser/configurator/projects/${projectId}/config/timeline-graph-grants`);
	}

	saveTimelineGraphGrants(projectId: string, grants: Record<string, string[]>): Observable<void> {
		return this.http.put<void>(`/api/superuser/configurator/projects/${projectId}/config/timeline-graph-grants`, grants);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/TimelineGraphGrantsService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface TimelineGraphGrantsService {

	Map<UUID, List<UUID>> getTimelineGraphGrants(UUID projectId);

	void saveTimelineGraphGrants(UUID projectId, Map<UUID, List<UUID>> profileTimelineGraphMap);
}
